# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1483391886 18000
#      Mon Jan 02 16:18:06 2017 -0500
# Node ID f71191aea679d2ffc158a7a463f7610f07fc03a0
# Parent  698c1190d79027fb1edc911ef7d3f224d20c3a41
provide macro declaring existence of f77 integer type

* oct-conf-post.in.h (OCTAVE_HAVE_F77_INT_TYPE): New macro
* mk-octave-config-h.sh: Write definition of OCTAVE_HAVE_F77_INT_TYPE
to octave-config.h.

diff --git a/build-aux/mk-octave-config-h.sh b/build-aux/mk-octave-config-h.sh
--- a/build-aux/mk-octave-config-h.sh
+++ b/build-aux/mk-octave-config-h.sh
@@ -160,16 +160,18 @@ if test -z "$octave_f77_int_type"; then
   exit 1
 fi
 
 cat << EOF
 
 typedef $octave_idx_type octave_idx_type;
 typedef $octave_f77_int_type octave_f77_int_type;
 
+#  define OCTAVE_HAVE_F77_INT_TYPE 1
+
 EOF
 
 if grep "#define OCTAVE_ENABLE_FLOAT_TRUNCATE 1" $config_h_file > /dev/null; then
   echo "#  define OCTAVE_FLOAT_TRUNCATE volatile"
 else
   echo "#  define OCTAVE_FLOAT_TRUNCATE"
 fi
 
diff --git a/oct-conf-post.in.h b/oct-conf-post.in.h
--- a/oct-conf-post.in.h
+++ b/oct-conf-post.in.h
@@ -180,14 +180,16 @@ typedef unsigned long ino_t;
 #  include <cstdint>
 #else
 #  include <stdint.h>
 #endif
 
 typedef OCTAVE_IDX_TYPE octave_idx_type;
 typedef OCTAVE_F77_INT_TYPE octave_f77_int_type;
 
+#define OCTAVE_HAVE_F77_INT_TYPE 1
+
 /* Tag indicating Octave's autoconf-generated config.h has been
    included.  This symbol is provided because autoconf-generated
    config.h files do not define a multiple-inclusion guard.  See also
    the notes at the top of the generated octave-config.h file.  */
 
 #define OCTAVE_AUTOCONFIG_H_INCLUDED 1
